rq/queue.py

```diff
@@ -754,8 +758,10 @@ def enqueue_many(self, job_datas: List['EnqueueData'], pipeline: Optional['Pipel
             List[Job]: A list of enqueued jobs
         """
         pipe = pipeline if pipeline is not None else self.connection.pipeline()
-        jobs = [
-            self._enqueue_job(
+        if any([job_data.depends_on for job_data in job_datas]):
```

Comment:
I think we should split the jobs into two different lists, with and without dependencies.

Reply:
I tried this but it got pretty messy. It seems cleaner to just pass in all jobs to `Dependency.get_ready_jobs` and let that function return the jobs without dependencies, along with those whose dependencies have been met.

Reply:
But this would mean if I enqueue 10 jobs and only one of them had dependencies, the 9 other jobs would be created in deferred state to be requeued later, right? I'd rather we do something like this:
```python
jobs_with_dependencies = [job_data for job_data in job_datas if job_data.depends_on]
if jobs_with_dependencies:
    ...
jobs_without_dependencies = [job_data for job_data in job_datas if not job_data.depends_on]
if jobs_without_dependencies:
    ....
```

Reply:
Makes sense, just pushed this change